tensorflow/core/kernels/reduction_gpu_kernels.cu.h

```diff
@@ -510,10 +508,9 @@ __global__ void ColumnReduceInToTempKernel(void* __restrict__ temp,
 }
 
 template <typename T, typename OUT_T, typename Op>
-__global__ void ColumnReduceTempToOutKernel(void* __restrict__ temp,
-                                            int temp_in_offset, T in, OUT_T out,
-                                            int num_planes, int num_rows,
-                                            int num_cols, Op op) {
+__global__ __launch_bounds__(1024) void ColumnReduceTempToOutKernel(
```

Comment:
ROCm 3.5+ (more specifically the hip-clang compiler) assumes a default value of 256 (for max threads per block) for GPU kernels, in case where that value is not explicitly specified via the `__launch_bounds__ `attribute. 

If such a kernel (which has no `__launch_bounds__ `attribute specified) is called at runtime with a threads_per_block value greater than 256, it is possible to run into some undetermined behaviour. This will be changed to become a runtime error instead (in a forthcoming ROCm release).

Reply:
I see. Thanks for the explanation.